book/optimization.qmd

```diff
@@ -1090,34 +1089,111 @@ filter = flt("importance", learner = lrn)
 filter$calculate(task)
 head(as.data.table(filter), 3)
 ```
+### Filter-based Feature Selection
+
+A common usecase for filters is to conduct feature selection based on the filter scores.
+This can be achived using `r ref("mlr_pipeops_filter", text = "PipeOpFilter")`.
+This PipeOp takes as input a Task, applies the filter, and selects the features based on the calculated scores.
+The method how the features are subset can be defined in four different ways, each corresponding to a different parameter:
+
+  * `filter.nfeat` - The *nfeat* values with the highest filter score are selected.
+  * `filter.frac` - The *fraction* of values with the highest filter score are selected.
+  * `filter.cutoff` - All features with a filter score greater than the specified *cutoff* are selected.
+  * `filter.permuted` - A random permutation of each feature is added to the task before applying the filter.
+    All features selected before the *permuted*-th permuted features is selected are kept.
+
+
+:::{.callout-note}
+The `r ref("mlr_pipeops_filter", text = "PipeOpFilter")` only operates on the subset of features that have the correct column type and will leave other columns as is.
+:::
+
+We will illustrate the usage of this filter with the use of a machine learning pipeline for the `r ref("mlr_tasks_optdigits", text = "optdigits")` task, where the goal is to classify handwritten digits.
+We will first subset the features based on the `r ref("mlr3filters::mlr_filters_information_gain", text = "FilterInformationGain")` and then fit a `r ref("mlr3learners::mlr_learners_classif.lda", text = "LDA")`.
+We will tune the parameter `filter.frac` with a simple grid search and visualize the classification error for the different fractions.
```

Comment:
At this point it would be good to mention that the tuning of filter hyperpars can be fusioned with learner hyperpar tuning. This is in fact one of the big advantages of filters: they don't need an extra layer of tuning (as wrappers do).

Maybe even worth putting this into a "tip" block.